engine/src/main/java/org/kigalisim/engine/state/YearMatcher.java

```diff
@@ -13,7 +13,11 @@
  * Class representing a range of years where inclusion can be tested.
  *
  * <p>Supports testing year inclusion within a range where null values
- * represent positive or negative infinity bounds.</p>
+ * represent positive or negative infinity bounds. This implementation is
+ * functionally equivalent to the JavaScript YearMatcher, with the main 
```

Comment:
This is great! However, please remove the mention of the difference to JS. We will document this elsewhere.